File: PassGen.py
# Password generator that takes advantage of the idea that 4 random
# words would provide better security than a single word employing
# 'l33t speak'. This generator will also employ l33t speak along with
# the use of 4 words, as opposed to the Java version that does not.
import random
gen = ""
for i in range(4):
    file = open("dict.txt")
    for j in range(random.randint(0, 10000)):
        read = file.readline()
    gen += read
gen = gen.replace('\n','_');

#time for some leet speak lols

gen = gen.replace('s','$');
gen = gen.replace('S','$');

gen = gen.replace('a','@');
gen = gen.replace('A','4');

gen = gen.replace('i','1');
gen = gen.replace('I','1');

gen = gen.replace('o','0');
gen = gen.replace('O','0');

gen = gen.replace('e','3');
gen = gen.replace('E','3');

print("gen: \"" + gen + "\", Length: " + str(len(gen)))

if len(gen) < 25:
    print("\tThe password generated may be too short")

File: PassGen.java
/**
 * Password generator that takes advantage of the idea that 4 random
 * words would provide better security than a single word employing
 * 'l33t speak'.
 */
import java.io.*;
import java.util.Random;
public class PassGen {
	public static void main(String args[]) throws IOException {
		BufferedReader in = null;
		Random ran = new Random();
		String gen = "";
		try {
			String check = "";
			
			//Chooses 4 words from a list of 10000 words in file dict.txt
			for (int i = 0; i < 4; i++) {
				
				//Chooses one of 10000 words
				int rand = ran.nextInt(10000);
				in = new BufferedReader(new FileReader("dict.txt"));
				
				//searches for the word
				for (int j = 0; j < rand; j++) {
					check = in.readLine();
				}
				gen += check;
				gen += "_";
			}
		}
		finally {
			if (in != null) {
				in.close();
			}
		}
		System.out.println("gen: \"" + gen + "\", Length: " + gen.length());
		if (gen.length() < 25) {
			//Generally, a 25 characters is the minimum number of characters I accept as secure
			System.err.println("\tThe password generated may be too short");
		}
	}
}